packages/legend-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -277,6 +348,23 @@ export const getRootMappingData = (
   };
 };
 
+const generateExplorerTreeClassNodeChildrenIDs = (
+  node: QueryBuilderExplorerTreeNodeData,
+): string[] => {
+  const currentClass = node.type as Class;
+  const idsFromProperties = (
+    node instanceof QueryBuilderExplorerTreeSubTypeNodeData
+      ? currentClass.getAllOwnedProperties()
+      : currentClass
+          .getAllProperties()
+          .concat(currentClass.getAllDerivedProperties())
+  ).map((p) => `${node.id}.${p.name}`);
+  const idsFromSubClasses = currentClass._subClasses.map(
+    (subClass) => `${node.id}->subType(${TYPE_CAST_TOKEN}${subClass.path})`,
```

Comment:
Can we tokenize `->`, also `subType` should be called with `extractElementNameFromPath(SUPPORTED_FUNCTION.SUBTYPE)`

Reply:
Also, is there a reason why we must use the Pure grammar for the subType ID format? I have the feeling that if we do this, it will trick us into thinking that the ID is important in the logic somehow. ID is meant to be identifier, so let's keep it far from something useful logic-wise, for example, do `${node.id}@${subClass.path}`